thirdparty/github.go

```diff
@@ -60,72 +60,110 @@ var (
 	GithubPatchBaseRepoKey  = bsonutil.MustHaveTag(GithubPatch{}, "BaseRepo")
 )
 
-func githubShouldRetry(index int, req *http.Request, resp *http.Response, err error) bool {
-	if index >= NumGithubAttempts {
-		return false
-	}
+func githubShouldRetry(caller string) utility.HTTPRetryFunction {
+	return func(index int, req *http.Request, resp *http.Response, err error) bool {
+		if index >= NumGithubAttempts {
+			return false
+		}
 
-	url := req.URL.String()
+		url := req.URL.String()
 
-	if err != nil {
-		temporary := utility.IsTemporaryError(err)
-		grip.Error(message.WrapError(err, message.Fields{
-			"message":   "failed trying to call github",
-			"method":    req.Method,
-			"url":       url,
-			"temporary": temporary,
-		}))
-		return temporary
-	}
+		if err != nil {
+			temporary := utility.IsTemporaryError(err)
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":   "failed trying to call github",
+				"method":    req.Method,
+				"url":       url,
+				"temporary": temporary,
+			}))
+			if temporary {
+				grip.Info(message.Fields{
```

Comment:
nit `grip.InfoWhen` instead of `if temporary`

Reply:
classic